tests/pytests/test_hybrid_internal.py

```diff
@@ -0,0 +1,279 @@
+    result = env.cmd('_FT.HYBRID', 'idx', 'SEARCH', '@description:running',
+                     'VSIM', '@embedding', '$vec_param', 'WITHCURSOR', 'DIALECT', '2',
+                     'PARAMS', '2', 'vec_param', query_vec.tobytes())
+
+    # Should return a map with VSIM and SEARCH cursor IDs
+    env.assertTrue(isinstance(result, list))
+    env.assertTrue(len(result) > 0)
+
+    # Convert list to dict for easier access
+    result_dict = dict(zip(result[::2], result[1::2]))
+
+    # Should have VSIM and SEARCH cursor IDs
+    env.assertIn('VSIM', result_dict)
+    env.assertIn('SEARCH', result_dict)
+
+    # Both cursor IDs should be valid integers
+    vsim_cursor = result_dict['VSIM']
+    search_cursor = result_dict['SEARCH']
+    env.assertTrue(isinstance(vsim_cursor, (int, str)))
+    env.assertTrue(isinstance(search_cursor, (int, str)))
+
+
+@skip(cluster=True)
+def test_hybrid_internal_with_count_parameter(env):
+    """Test _FT.HYBRID with WITHCURSOR and COUNT parameter"""
+    setup_hybrid_test_data(env)
+
+    # Execute with COUNT parameter set to 2
+    count_param = 2
+    query_vec = create_np_array_typed([0.0, 0.0], 'FLOAT32')
+    result = env.cmd('_FT.HYBRID', 'idx', 'SEARCH', '@description:running',
+                     'VSIM', '@embedding', '$vec_param', 'WITHCURSOR', 'COUNT', str(count_param), 'DIALECT', '2',
+                     'PARAMS', '2', 'vec_param', query_vec.tobytes())
+
+    # Should return a map with cursor IDs
+    env.assertTrue(isinstance(result, list))
+    result_dict = dict(zip(result[::2], result[1::2]))
+
+    # Should have both cursor types
+    env.assertIn('VSIM', result_dict)
+    env.assertIn('SEARCH', result_dict)
+
+    # Test reading from cursors with COUNT parameter using callback
+    def validate_batch_size(batch_results, cursor_response):
+        """Callback to validate that each batch respects the COUNT parameter"""
+        # The key test: number of results in each batch should be <= COUNT parameter
+        env.assertTrue(len(batch_results) <= count_param)
+
+    for cursor_id in result_dict.values():
+        if cursor_id != 0:  # Only test active cursors
+            # Use common function with callback to validate COUNT behavior
+            results = read_cursor_completely(env, 'idx', cursor_id, validate_batch_size)
+            env.assertTrue(isinstance(results, list))
+
+
+@skip(cluster=True)
+def test_hybrid_internal_cursor_interaction(env):
+    """Test reading from both VSIM and SEARCH cursors and compare with equivalent FT.SEARCH commands"""
+    setup_hybrid_test_data(env)
+
+    # Execute the hybrid command with cursors
+    query_vec = create_np_array_typed([1.0, 0.0], 'FLOAT32')
+    hybrid_result = env.cmd('_FT.HYBRID', 'idx', 'SEARCH', '@description:shoes',
+                           'VSIM', '@embedding', '$vec_param', 'WITHCURSOR', 'DIALECT', '2',
+                           'PARAMS', '2', 'vec_param', query_vec.tobytes())
```

Comment:
Lets trim the query a bit

```suggestion
    hybrid_result = env.cmd('_FT.HYBRID', 'idx', 'SEARCH', '@description:shoes',
                           'VSIM', '@embedding', query_vec.tobytes(), 'WITHCURSOR')
```